src/Parser/Hunter/Marksmanship/Modules/Talents/TrueAim.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox from 'Main/StatisticBox';
+
+class TrueAim extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+  };
+
+  _currentStacks = 0;
+  startOfMaxStacks = 0;
+  timeAtMaxStacks = 0;
+
+  //starts -1 because the stacks drop at end of combat, but that shouldn't count as a time where it dropped
+  timesDropped = -1;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.TRUE_AIM_TALENT.id);
+  }
+
+  on_byPlayer_applydebuffstack(event) {
+    const debuffId = event.ability.guid;
+    if (debuffId !== SPELLS.TRUE_AIM_DEBUFF.id) {
+      return;
+    }
+    //I can't use on_byPlayer_applydebuff as it won't work in conjunction with on_byplayer_applydebuffstack
+    //so we do it this way, and just make a check if _currentStacks is 0, in which case we set it to 1 to indicate it already has been applied.
+    if (this._currentStacks === 0) {
+      this._currentStacks = 1;
+    }
```

Comment:
Interesting. I guess this would work but why it wouldn't work with applydebuffstack? Just out of curiosity.

Reply:
Because they happen in the "wrong" order for that to be possible: 
![image](https://user-images.githubusercontent.com/29204244/31851925-b3a7d9e8-b66f-11e7-8cec-234c74f63ec2.png)

I see I wrote "in conjunction with on_byplayer_applydebuffstack" , but I actually meant "in conjunction with on_byplayer_removedebuff"